src/datasource.js

```diff
@@ -1,14 +1,17 @@
 import _ from 'lodash';
-import { convertFilterValueToProperType, ATTRIBUTE } from './types/queryAttributes';
+import { convertFilterValueToProperType, ATTRIBUTE, METRICS_ATTRIBUTE_LIST } from './types/queryAttributes';
 import { AGGREGATION } from './types/aggregations';
 import { calculateAutoInterval, QUERY_INTERVAL } from './types/intervals';
 import { transform } from './result_transformer';
 import { ResultFormat } from './types/resultFormat';
 
-const getApiRequestUrl = (baseUrl, isAdAnalytics) => {
+const getApiRequestUrl = (baseUrl, isAdAnalytics, isMetric) => {
   if (isAdAnalytics === true) {
     return baseUrl + '/analytics/ads/queries';
   }
+  if (isMetric == true) {
```

Comment:
What happens if the user configured an `Ad` datasource and selects a `metric`? :)
The metric `baseUrl` should be prioritised over an ad `baseUrl` imo.